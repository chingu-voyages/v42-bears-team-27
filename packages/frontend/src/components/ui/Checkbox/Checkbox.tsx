/* eslint-disable jsx-a11y/label-has-associated-control */
import type { RefAttributes, ForwardRefExoticComponent } from 'react';
import { forwardRef } from 'react';
import type { ThemeUIStyleObject } from 'theme-ui';
import * as CheckboxPrimitive from '@radix-ui/react-checkbox';
import { MdCheck } from 'react-icons/md';

export interface CheckboxProps extends CheckboxPrimitive.CheckboxProps {
  /**
   * Label for the component
   */
  label?: string;
  /**
   * Size of the component
   * @default 18
   */
  size?: number;
  sx?: ThemeUIStyleObject;
}

export const Checkbox: ForwardRefExoticComponent<
CheckboxProps & RefAttributes<HTMLButtonElement>
> = forwardRef(({ children, key, id, label, size = 18, sx, ...rest }, ref) => (
  <div
    key={key}
    sx={{ display: 'flex', alignItems: 'center', color: 'primary', ...sx }}
  >
    <CheckboxPrimitive.Root
      ref={ref}
      id={id}
      sx={{
        variant: 'buttons.primary.outlined',
        p: 0,
        width: `calc(${size}px + 4px)`,
        height: `calc(${size}px + 4px)`,
        cursor: 'pointer',
      }}
      {...rest}
    >
      <CheckboxPrimitive.Indicator>
        <MdCheck size={size} />
      </CheckboxPrimitive.Indicator>
    </CheckboxPrimitive.Root>
    {label && (
      <label
        htmlFor={id}
        sx={{
          variant: 'text.label',
          ml: 2,
          fontFamily: 'heading',
          fontSize: 1,
          fontWeight: 'regular',
          lineHeight: 'heading',
        }}
      >
        {label}
      </label>
    )}
    {children}
  </div>
));

if (process.env.NODE_ENV !== 'production') {
  Checkbox.displayName = 'Checkbox';
}
