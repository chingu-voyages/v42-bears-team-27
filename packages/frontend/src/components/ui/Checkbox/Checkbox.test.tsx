import { render } from 'test-utils';

import { Checkbox } from './Checkbox';

describe('<Checkbox />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Checkbox />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with label prop equals Foo', () => {
      const { container } = render(<Checkbox label="Foo" />);
      expect(container).toMatchSnapshot();
    });
  });
});
