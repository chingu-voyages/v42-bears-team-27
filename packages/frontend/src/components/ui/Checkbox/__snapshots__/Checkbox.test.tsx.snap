// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Snapshots should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--theme-ui-colors-primary);
}

.emotion-1 {
  background-color: white;
  color: var(--theme-ui-colors-primary);
  border: 2px solid currentColor;
  padding: 0;
  width: calc(18px + 4px);
  height: calc(18px + 4px);
  cursor: pointer;
}

.emotion-1:hover {
  background-color: #e6e6e6;
}

<div>
  <div
    class="emotion-0"
  >
    <button
      aria-checked="false"
      class="emotion-1"
      data-state="unchecked"
      role="checkbox"
      type="button"
      value="on"
    />
  </div>
</div>
`;

exports[`<Checkbox /> Snapshots should match snapshot with label prop equals Foo 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--theme-ui-colors-primary);
}

.emotion-1 {
  background-color: white;
  color: var(--theme-ui-colors-primary);
  border: 2px solid currentColor;
  padding: 0;
  width: calc(18px + 4px);
  height: calc(18px + 4px);
  cursor: pointer;
}

.emotion-1:hover {
  background-color: #e6e6e6;
}

.emotion-2 {
  font-family: "Roboto",sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.2;
  margin-left: 8px;
}

<div>
  <div
    class="emotion-0"
  >
    <button
      aria-checked="false"
      class="emotion-1"
      data-state="unchecked"
      role="checkbox"
      type="button"
      value="on"
    />
    <label
      class="emotion-2"
    >
      Foo
    </label>
  </div>
</div>
`;
